src/Telegram.Bot.Core/ITelegramBotClient.cs

```diff
@@ -119,19 +62,8 @@ public interface ITelegramBotClient
         /// <returns><c>true</c> if token is valid</returns>
         Task<bool> TestApiAsync(CancellationToken cancellationToken = default);
 
-        /// <summary>
-        /// Start update receiving
-        /// </summary>
-        /// <param name="allowedUpdates">List the types of updates you want your bot to receive.</param>
-        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
-        /// <exception cref="Exceptions.ApiRequestException"> Thrown if token is invalid</exception>
-        void StartReceiving(UpdateType[] allowedUpdates = null,
-                            CancellationToken cancellationToken = default);
-
-        /// <summary>
-        /// Stop update receiving
-        /// </summary>
-        void StopReceiving();
+        /// <inheritdoc cref="ITelegramBotJsonConverter"/>
+        ITelegramBotJsonConverter JsonConverter { get; set; }
```

Comment:
I don't think it should have public setter. JsonConverter can be accessed concurrently from more than one thread so there's a possibility of getting null reference exception if one thread attempts to overwrite it.

Reply:
We can just notify everyone that this property is not thread safe. And we can implement something like ConcurrentTelegramBotClient which will be completely thread safe.